### PR TITLE
Feature/models article tag

### PR DESCRIPTION
**Pull request by olcplif**
- Add models Article and Tag.
- Create a relationship between Article and Tag (through Tagging).
- Created a relationship between Article and User (author).

Trello task - [https://trello.com/c/vRmnMM8n/14-create-models-article-articletag](https://trello.com/c/vRmnMM8n/14-create-models-article-articletag)

### CODE REVIEW

**Review by bagiratania98 — APPROVED** [review]
LGTM!

File: app/models/article.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Article < ApplicationRecord
+  include PGEnum(status: %w[draft published trashed])
```
Comment (mmrshk) [inline app/models/article.rb:2]: Oh. no-no, we are not doing like this - https://betterprogramming.pub/how-to-use-enums-in-rails-6-87600e292476

File: Gemfile [hunk 2]
```diff
@@ -7,6 +7,8 @@ ruby '3.0.4'
 gem 'rails', '~> 6.0.5'
 # Use sqlite3 as the database for Active Record
 gem 'pg'
+# Gem for use enum
+gem 'activerecord-pg_enum'
```
Comment (mmrshk) [inline Gemfile:11]: delete this gem

File: app/models/tag.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Tag < ApplicationRecord
+  has_many :taggings, dependent: :destroy
+  has_many :articles, through: :taggings
+
+  validates :name, presence: true, length: { minimum: 2 }
+end
```
Comment (mmrshk) [inline app/models/tag.rb:6]: empty line

File: app/models/tagging.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Tagging < ApplicationRecord
```
Comment (mmrshk) [inline app/models/tagging.rb:1]: rename to ArticleTag

File: db/migrate/20220629173526_create_articles.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+class CreateArticles < ActiveRecord::Migration[6.0]
+
```
Comment (mmrshk) [inline db/migrate/20220629173526_create_articles.rb:2]: redundant line

File: db/migrate/20220629173548_create_tags.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class CreateTags < ActiveRecord::Migration[6.0]
+  def change
+    create_table :tags do |t|
+      t.string :name
```
Comment (mmrshk) [inline db/migrate/20220629173548_create_tags.rb:4]: null: false

File: spec/models/article_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Article, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (mmrshk) [inline spec/models/article_spec.rb:4]: pending? 🙂

File: app/models/tag.rb (hunk 3 above)
Comment (olcplif) [reply app/models/tag.rb:5]: fixed

File: app/models/tagging.rb (hunk 4 above)
Comment (olcplif) [reply app/models/tagging.rb]: done

File: db/migrate/20220629173526_create_articles.rb (hunk 5 above)
Comment (olcplif) [reply db/migrate/20220629173526_create_articles.rb:2]: done

File: db/migrate/20220629173548_create_tags.rb (hunk 6 above)
Comment (olcplif) [reply db/migrate/20220629173548_create_tags.rb:4]: fixed

File: spec/models/article_spec.rb (hunk 7 above)
Comment (olcplif) [reply spec/models/article_spec.rb:4]: This is a generated phrase :) But I will add some tests

File: app/models/article.rb (hunk 1 above)
Comment (olcplif) [reply app/models/article.rb]: Thx! It is much simpler. Wonder why it didn't work for me the first time...

File: Gemfile (hunk 2 above)
Comment (olcplif) [reply Gemfile]: Gem 'activerecord-pg_enum' deleted!

File: app/helpers/article_helper.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module ArticleHelper
+  VALID_STATUSES = {
```
Comment (igorkasyanchuk) [inline app/helpers/article_helper.rb:4]: why here? constants, put in model

File: spec/models/article_spec.rb [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      text: 'Some text for First Article',
+      user_id: 1,
+      status: 'published',
+      language: 'eng'
+    )
+    expect(article.text).to eq('Some text for First Article')
+  end
+
+  it 'returns status' do
+    article = Article.create(
```
Comment (igorkasyanchuk) [inline spec/models/article_spec.rb:29]: add factory_bot + faker gems

CC @mmrshk

File: app/helpers/article_helper.rb (hunk 8 above)
Comment (olcplif) [reply app/helpers/article_helper.rb]: Constants have been moved to the model. I thought it was a good idea to keep the models thin :)

File: app/helpers/article_helper.rb [hunk 10]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+module ArticleHelper
+end
```
Comment (viktoriia-horobets) [inline app/helpers/article_helper.rb:4]: please, remove empty helper

Comment (olcplif) [reply app/helpers/article_helper.rb]: Empty helper removed

File: spec/models/article_spec.rb (hunk 9 above)
Comment (olcplif) [reply spec/models/article_spec.rb]: I added these gems. And I will rebuild the rspec tests.

**Pull request merged**